### PR TITLE
Defined dynamoose transaction in dynamoose.d.ts

### PR DESCRIPTION
**Pull request by yoonhoGo**
### Summary:
I defined a dynamoose transaction in the dynamoose.d.ts file.
I created it at a level that matches the attributes shown in the documentation.

### Type (select 1):
- [ ] Bug fix
- [ ] Feature implementation
- [x] Documentation improvement
- [ ] Testing improvement

- [ ] Test added to report bug (GitHub issue #--- @---)
- [ ] Something not listed here

### Is this a breaking change? (select 1):
- [ ] 🚨 YES 🚨
- [x] No
- [ ] I'm not sure

### Is this ready to be merged into Dynamoose? (select 1):
- [x] Yes
- [ ] No

### Are all the tests currently passing on this PR? (select 1):
- [x] Yes
- [ ] No

### Other:
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [x] I have updated the Dynamoose documentation (if required) given the changes I made
- [x] I have added/updated the Dynamoose test cases (if required) given the changes I made
- [x] I have run `npm test` from the root of the project directory to ensure all tests continue to pass
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamoose license](https://github.com/dynamoosejs/dynamoose/blob/master/LICENSE.txt)
- [x] All of my commits and commit messages are detailed, explain what changes were made, and are easy to follow and understand
- [x] I have confirmed that all my code changes are indented properly using 2 spaces
- [x] I have filled out all fields above

### CODE REVIEW

File: dynamoose.d.ts [hunk 1]
```diff
@@ -16,6 +16,18 @@ declare module "dynamoose" {
   export function setDefaults(options: ModelOption): void;
   export function setDDB(ddb: _AWS.DynamoDB): void;
   export function revertDDB(): void;
+  export function transaction<DataSchema, KeySchema>(
+    items: Array<
+      Promise<ModelSchema<DataSchema>>
```
Comment (fishcharlie) [inline dynamoose.d.ts]: I don't think this is correct. `dynamoose.transaction` doesn't accept a promise in the array.

File: dynamoose.d.ts [hunk 2]
```diff
@@ -16,6 +16,18 @@ declare module "dynamoose" {
...
+      | _AWS.DynamoDB.TransactWriteItem
+      | _AWS.DynamoDB.TransactGetItem
+    >,
+    options?: TransactionOptions,
+    next?: any
```
Comment (fishcharlie) [inline dynamoose.d.ts]: I don't think this is correct. This is a callback, not an `any`. See the `get` example I found in this file.

```js
get(key: KeySchema, callback?: (err: Error, data: DataSchema) => void): Promise<ModelSchema<DataSchema> | undefined>;
```

File: dynamoose.d.ts [hunk 3]
```diff
@@ -306,4 +320,32 @@ declare module "dynamoose" {
     get(fn: any): any;
     set(fn: any): any;
   }
+
+  /**
+   * Transaction
+   */
+  export interface ModelTransactionConstructor<DataSchema, KeySchema> {
```
Comment (fishcharlie) [inline dynamoose.d.ts]: `Model.transaction` methods don't return promises or have callbacks.

File: dynamoose.d.ts [hunk 4]
```diff
@@ -306,4 +320,32 @@ declare module "dynamoose" {
...
+
+    delete(key: KeySchema, callback?: (err: Error) => void): Promise<undefined>;
+
+    update(key: KeySchema, update: UpdateUpdate<DataSchema>, options: UpdateOption, callback: (err: Error, items: ModelSchema<DataSchema>[]) => void): void;
+    update(key: KeySchema, update: UpdateUpdate<DataSchema>, callback: (err: Error, items: ModelSchema<DataSchema>[]) => void): void;
+    update(key: KeySchema, update: UpdateUpdate<DataSchema>, options?: UpdateOption): Promise<ModelSchema<DataSchema>>;
+
+    conditionCheck(key: KeySchema, options?: ConditionOptions): void
+  }
+  export interface ConditionOptions {
```
Comment (fishcharlie) [inline dynamoose.d.ts]: Does this not already exist in this file? Conditions were allowed in other methods before this.

File: dynamoose.d.ts (hunk 3 above)
Comment (yoonhoGo) [reply dynamoose.d.ts]: Reference line 222, 241 in `lib/index.js`.
first paramater `items` of `dynamoose.transaction` is used `await Promise.all(items);`

File: dynamoose.d.ts (hunk 1 above)
Comment (yoonhoGo) [reply dynamoose.d.ts]: Reference line 222, 241 in lib/index.js.
First paramater items of dynamoose.transaction is used await Promise.all(items);

**Pull request merged**